engine/src/main/java/org/apache/apex/stram/ContainerRedeployRequest.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.stram;
+
+import java.util.Set;
+
+import com.google.common.collect.Sets;
+
+public class ContainerRedeployRequest
+{
+  private long groupId;
+  private Set<Integer> opeartorsToDeploy = Sets.newHashSet();
+  private Set<Integer> opeartorsToUndeploy = Sets.newHashSet();
+  private Set<String> affectedContainers = Sets.newHashSet();
```

Comment:
Can we have Set<PTContainer> instead? Or we can have both Set<String> and Set<PTContainer> if you want easy lookup by external containerId (since the String is the externalContainerId, right?). The advantage is that PTContainer is the "logical" container object that gets redeployed but gets a different externalId and so we don't need to remove and add a string in this request. That will help in simplifying and understanding the logic.